pkg/migrate/migrate.go

```diff
@@ -987,6 +997,26 @@ func swapPVs(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
 	return nil
 }
 
+// waitForDeletion waits for the provided pvcName to not be found, and returns early if any error besides 'not found' is given
+func waitForDeletion(ctx context.Context, clientset k8sclient.Interface, pvcName, ns string) error {
+	for true {
+		_, err := clientset.CoreV1().PersistentVolumeClaims(ns).Get(ctx, pvcName, metav1.GetOptions{})
+		if k8serrors.IsNotFound(err) {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		select {
+		case <-time.After(time.Second / 20):
+			continue
+		case <-ctx.Done():
+			return fmt.Errorf("context ended waiting for PVC %s in %s to be deleted", pvcName, ns)
+		}
```

Comment:
Do you think we need a timeout case here? Maybe 5m?

Reply:
Hmm, maybe

Reply:
I think nothing else in this project has timeouts yet - but I should really add logs to show what it's waiting for